plugins/stringlib/stringlib.cpp

```diff
@@ -223,6 +229,270 @@ unsigned editDistance(unsigned leftLen, const char * left, unsigned rightLen, co
     return da[mask(leftLen-1)][rightLen-1];
 }
 
+unsigned optimalStringAlignmentDistance(unsigned leftLen, const char * left, unsigned rightLen, const char * right)
+{
+    /* This is the restricted form of the Damerau-Levenshtein distance AKA optimal string alignment (OSA).
+     * It is restricted in that a substring can only be altered once.
+     * e.g. OSA(CA, ABC) = 3, CA -> A -> AB -> ABC : fullDL(CA, ABC) = 2, CA -> AC -> ABC (i.e. a transposition then an insertion upon that transposition).
+     * e.g. OSA(abcde, acbed) = 3 & fullDL(abcde, acbed) = 2
+     */
+
+    unsigned i, j;
+
+    clip(leftLen, left);
+    clip(rightLen, right);
+
+    if (rightLen < 2 || leftLen < 2)
+        return editDistance(leftLen, left, rightLen, right);
+
+    if (leftLen > 255)
+        leftLen = 255;
+
+    if (rightLen > 255)
+        rightLen = 255;
+
+    if (leftLen == 0)
+        return rightLen;
+
+    if (rightLen == 0)
+        return leftLen;
+
+#if 0
```

Comment:
This code should be deleted, or moved/simplified in a comment if it is being retained for reference.